angular/index.d.ts

```diff
@@ -1035,13 +1035,14 @@ declare namespace angular {
          *
          * @param reason Constant, message, exception or an object representing the rejection reason.
          */
-        reject(reason?: any): IPromise<any>;
+        reject(reason?: any): IPromise<never>;
         /**
          * Wraps an object that might be a value or a (3rd party) then-able promise into a $q promise. This is useful when you are dealing with an object that might or might not be a promise, or if the promise comes from a source that can't be trusted.
          *
          * @param value Value or a promise
          */
         resolve<T>(value: IPromise<T>|T): IPromise<T>;
+        resolve<T1, T2>(value: IPromise<T1>|T2): IPromise<T1|T2>;
```

Comment:
1. It allows to handle rare edge cases. I agree that it is not very useful, but still there are "legal" cases that fail to compile with vanilla definitions of Promises:

```
Promise.resolve(Math.random() > 0.5 ? 2 : Promise.reject(2))
    .then(x => console.log(x), x => console.error(x));
```

2. On other hand, I agree that better if my pull request will target `lib.es2015.promise.d.ts`. Consistency across the typings is a valuable thing.